src/com/mendix/widget/carousel/components/__tests__/Carousel.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { ShallowWrapper, shallow } from "enzyme";
+import { DOM, createElement } from "react";
+
+import { Carousel, CarouselProps, Image } from "../Carousel";
+import { CarouselItem } from "../CarouselItem";
+
+describe("Carousel", () => {
+    let images: Image[];
+    let carousel: ShallowWrapper<CarouselProps, any>;
+    let carouselWrapper: ShallowWrapper<CarouselProps, any>;
+
+    it("renders the structure correctly", () => {
+        images = [ { url: "https://www.google.com/images/nav_logo242.png" } ];
+        carousel = shallow(createElement(Carousel, { images }));
+
+        expect(carousel).toBeElement(
+            DOM.div({ className: "mx-carousel" },
+                DOM.div({ className: "mx-carousel-item-wrapper" },
+                    createElement(CarouselItem, { active: true, url: images[0].url })
+            )));
+    });
+
+    describe("with no images", () => {
+
+        beforeEach(() => carousel = shallow(createElement(Carousel)) );
+
+        it("renders no carousel items", () => {
+            const carouselItems = carousel.find(CarouselItem);
+
+            expect(carouselItems.length).toBe(0);
+        });
+    });
+
+    describe("with one image", () => {
+
+        beforeEach(() => {
+            images = [ { url: "https://www.google.com/images/nav_logo242.png" } ];
+            carousel = shallow(createElement(Carousel, { images }));
+        });
+
+        it("renders one carousel item", () => {
+            const carouselItems = carousel.find(CarouselItem);
+
+            expect(carouselItems.length).toBe(1);
+
+            const carouselItem = carouselItems.first();
+
+            expect(carouselItem.type()).toEqual(CarouselItem);
```

Comment:
No need to do this, the `find` call on line 42 already ensures that the type is correct.